### PR TITLE
OpenSearch: Add Similarity Search with Score

### PR DESCRIPTION
**Pull request by naveentatikonda**
### Description
Add `similarity_search_with_score` method for OpenSearch to return scores along with documents in the search results

### CODE REVIEW

File: langchain/vectorstores/opensearch_vector_search.py [hunk 1]
```diff
@@ -454,17 +475,18 @@ def similarity_search(
             raise ValueError("Invalid `search_type` provided as an argument")
 
         response = self.client.search(index=self.index_name, body=search_query)
-        hits = [hit["_source"] for hit in response["hits"]["hits"][:k]]
-        documents = [
-            Document(
-                page_content=hit[text_field],
-                metadata=hit
-                if metadata_field == "*" or metadata_field not in hit
```
[the remark this answers is not in the document]
Comment (naveentatikonda) [reply langchain/vectorstores/opensearch_vector_search.py:462]: added it back

File: tests/integration_tests/vectorstores/test_opensearch.py [hunk 2]
```diff
@@ -23,6 +23,18 @@ def test_opensearch() -> None:
     assert output == [Document(page_content="foo")]
 
 
+def test_similarity_search_with_score() -> None:
+    """Test similarity search with score using Approximate Search."""
+    docsearch = OpenSearchVectorSearch.from_texts(
+        texts, FakeEmbeddings(), opensearch_url=DEFAULT_OPENSEARCH_URL
```
Comment (dev2049) [inline tests/integration_tests/vectorstores/test_opensearch.py:29]: could we add some metadata just to make sure it's being correctly returned

Comment (naveentatikonda) [reply tests/integration_tests/vectorstores/test_opensearch.py]: Updated the test to include metadata

**Comment by dev2049** [comment]
looks great, thanks @naveentatikonda!

**Pull request merged**